clients/go/pkg/worker/jobPoller.go

```diff
@@ -16,117 +16,131 @@
 package worker
 
 import (
-	"context"
-	"fmt"
-	"github.com/camunda/zeebe/clients/go/v8/pkg/entities"
-	"github.com/camunda/zeebe/clients/go/v8/pkg/pb"
-	"google.golang.org/grpc/codes"
-	"google.golang.org/grpc/status"
-	"io"
-	"log"
-	"sync"
-	"time"
+    "context"
+    "fmt"
+    "github.com/camunda/zeebe/clients/go/v8/pkg/entities"
+    "github.com/camunda/zeebe/clients/go/v8/pkg/pb"
+    "google.golang.org/grpc/codes"
+    "google.golang.org/grpc/status"
+    "io"
+    "log"
+    "sync"
+    "time"
 )
 
 type jobPoller struct {
-	client         pb.GatewayClient
-	request        *pb.ActivateJobsRequest
-	requestTimeout time.Duration
-	maxJobsActive  int
-	pollInterval   time.Duration
-
-	jobQueue       chan entities.Job
-	workerFinished chan bool
-	closeSignal    chan struct{}
-	remaining      int
-	threshold      int
-	metrics        JobWorkerMetrics
-	shouldRetry    func(context.Context, error) bool
+    client              pb.GatewayClient
+    request             *pb.ActivateJobsRequest
+    requestTimeout      time.Duration
+    maxJobsActive       int
+    initialPollInterval time.Duration
+    pollInterval        time.Duration
+
+    jobQueue       chan entities.Job
+    workerFinished chan bool
+    closeSignal    chan struct{}
+    remaining      int
+    threshold      int
+    metrics        JobWorkerMetrics
+    shouldRetry    func(context.Context, error) bool
+
+    backoffSupplier BackoffSupplier
 }
 
 func (poller *jobPoller) poll(closeWait *sync.WaitGroup) {
-	defer closeWait.Done()
-
-	// initial poll
-	poller.activateJobs()
-
-	for {
-		select {
-		// either a job was finished
-		case <-poller.workerFinished:
-			poller.remaining--
-			poller.setJobsRemainingCountMetric(poller.remaining)
-		// or the poll interval exceeded
-		case <-time.After(poller.pollInterval):
-		// or poller should stop
-		case <-poller.closeSignal:
-			poller.setJobsRemainingCountMetric(0)
-			return
-		}
-
-		if poller.shouldActivateJobs() {
-			poller.activateJobs()
-		}
-	}
+    defer closeWait.Done()
+
+    // initial poll
+    poller.activateJobs()
+
+    for {
+        select {
+        // either a job was finished
+        case <-poller.workerFinished:
+            poller.remaining--
+            poller.setJobsRemainingCountMetric(poller.remaining)
+            poller.pollInterval = poller.initialPollInterval
```

Comment:
Only this line is new. I reset the pollInterval here after a successful job finish. 
```suggestion
            poller.pollInterval = poller.initialPollInterval
```